lib/src/ignore/files.rs

```diff
@@ -62,18 +64,26 @@ pub async fn from_origin(path: impl AsRef<Path>) -> (Vec<IgnoreFile>, Vec<Error>
 	match find_file(base.join(".git/config")).await {
 		Err(err) => errors.push(err),
 		Ok(None) => {}
-		Ok(Some(path)) => match git2::Config::open(&path) {
+		Ok(Some(path)) => match GitConfig::open(&path) {
 			Err(err) => errors.push(Error::new(ErrorKind::Other, err)),
 			Ok(config) => {
-				if let Ok(excludes) = config.get_path("core.excludesFile") {
-					discover_file(
-						&mut files,
-						&mut errors,
-						None,
-						Some(ProjectType::Git),
-						excludes,
-					)
-					.await;
+				if let Ok(excludes) = config.value::<Cow<[u8]>>("core", None, "excludesFile") {
+					match String::from_utf8(excludes.into_owned()) {
```

Comment:
I'm pretty sure you can go from a Cow<u8> to a Path without failing on non-utf8 paths, which is something I'd rather avoid.

Reply:
It's actually not as easy as it looks. The `git-config` crate has a Path type that can do that "automatically", but it has not been released in a stable version yet. I'll ping the maintainers to see when they are planning a release, and will update this accordingly.